snuba/query/conditions.py

```diff
@@ -313,40 +315,67 @@ def is_condition(exp: Expression) -> bool:
 
 
 def build_match(
-    col: str,
-    ops: Sequence[str],
-    param_type: Any,
+    col: Optional[str] = None,
+    subscriptable: Optional[str] = None,
+    ops: Optional[Sequence[str]] = None,
+    array_ops: Optional[Sequence[str]] = None,
+    param_type: Optional[Any] = None,
     alias: Optional[str] = None,
     key: Optional[str] = None,
 ) -> Or[Expression]:
-    # The IN condition has to be checked separately since each parameter
-    # has to be checked individually.
+    """
+    There is a common use case of matching a specific condition in our code base.
```

Comment:
This is better but not quite what I want as a reader.

What would make this easier are examples of usage and what patterns it would match (at least 2). 

Also, all the parameters are strings or `Any` and it's not clear what they mean unless I read the code in depth. A description of the parameters would be useful here too

Reply:
This is a pretty simple function that does one simple thing. I'm not sure what else I could add that makes it clearer. If you want examples of usage, look at examples of where we use it in the codebase. "Reading the code in depth" is less than 50 lines of code. That doesn't seem like huge burden.